packages/twenty-front/src/modules/ui/layout/resizable-panel/hooks/useResizablePanel.ts

```diff
@@ -0,0 +1,147 @@
+import { useCallback, useState } from 'react';
+
+import { useTrackPointer } from '@/ui/utilities/pointer-event/hooks/useTrackPointer';
+import { type PointerEventListener } from '@/ui/utilities/pointer-event/types/PointerEventListener';
+
+import { RESIZE_DRAG_THRESHOLD_PX } from '../constants/ResizeDragThresholdPx';
+import {
+  type ResizablePanelConstraints,
+  type ResizablePanelSide,
+} from '../types/ResizablePanelTypes';
+
+type UseResizablePanelProps = {
+  side: ResizablePanelSide;
+  constraints: ResizablePanelConstraints;
+  currentWidth: number;
+  onWidthChange: (width: number) => void;
+  onCollapse: () => void;
+  cssVariableName?: string;
+  onResizeStart?: () => void;
+};
+
+const clampWidth = (width: number, min: number, max: number): number =>
+  Math.min(max, Math.max(min, width));
+
+export const useResizablePanel = ({
```

Comment:
That's a pattern that is not adapted to a code base that use atomic state management, we avoid doing that and instead prefer setting recoil states synchronously from event handlers, and then components should listen to those states.